awesome/src/objects/tag.rs

```diff
@@ -34,8 +36,87 @@ impl Default for TagState {
 impl<'lua> Tag<'lua> {
     fn new(lua: &'lua Lua, args: Table) -> rlua::Result<Object<'lua>> {
         let class = class::class_setup(lua, "tag")?;
-        Ok(Tag::allocate(lua, class)?.handle_constructor_argument(args)?
-                                     .build())
+        Ok(object_setup(lua, Tag::allocate(lua, class)?)?
+                .handle_constructor_argument(args)?
+                .build())
+    }
+
+    pub fn get_clients(&self) -> rlua::Result<Table<'lua>> {
+        self.0.table()?.get("__clients")
+    }
+
+    pub fn set_clients(&mut self, clients: rlua::Table) -> rlua::Result<Value> {
+        // TODO: this is a really inefficient solution ( O(n*m) for the search )
```

Comment:
Yikes that is complicated...

Is there a reason we can't use [`std::HashSet`](https://doc.rust-lang.org/std/collections/struct.HashSet.html) to filter out the clients to add instead of this convulted logic?